### PR TITLE
Dashboard/overview outline

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @Tanmay-97 @lickem22 
Estimate: 20mins

---

## Ticket

Fixes: [AAQ-600](https://idinsight.atlassian.net/browse/AAQ-600)

### Goal

Setup layout for Overview page

Screenshot 2024-06-23 at 11 31 27 PM

### Changes

Just layout on front-end

### Future Tasks (optional)

See other tasks under [AAQ-582](https://idinsight.atlassian.net/browse/AAQ-582)
Basically, will start filling in components one by one.

Also, it is not mobile friendly so will need to do that at a later stage.

## How has this been tested?

Just clicking around front-end

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [ ] I have updated the automated tests
- [ ] I have updated the scripts in `scripts/`
- [ ] I have updated the requirements
- [ ] I have updated the README file
- [ ] I have updated affected documentation
- [ ] I have added a blogpost in Latest Updates
- [ ] I have updated the CI/CD scripts in `.github/workflows/`
- [ ] I have updated the Terraform code

[AAQ-600]: https://idinsight.atlassian.net/browse/AAQ-600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAQ-582]: https://idinsight.atlassian.net/browse/AAQ-582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: admin_app/src/app/dashboard/components/Overview.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React from "react";
```
Comment (sidravi1) [inline admin_app/src/app/dashboard/components/Overview.tsx:1]: I am starting to keep components next to the endpoint/page as per [this page](https://alexkondov.com/tao-of-react/).

We'll refactor the others to be the same at some stage

File: admin_app/src/components/NavBar.tsx [hunk 2]
```diff
@@ -28,12 +28,13 @@ const settings = ["Logout"];
 const NavBar = () => {
   return (
     <AppBar
-      position="static"
+      position="fixed"
```
Comment (sidravi1) [inline admin_app/src/components/NavBar.tsx:31]: I needed to make this change for the Drawer used for the Main Menu to not cover the navbar. This also meant I had to change the `layout.tsx` for other pages.

File: admin_app/src/app/dashboard/components/Sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import {
+  Drawer,
+  Box,
+  List,
+  ListItem,
+  ListItemText,
+  ListItemIcon,
+  ListItemButton,
+  Toolbar,
```
Comment (lickem22) [inline admin_app/src/app/dashboard/components/Sidebar.tsx:10]: Unused import

File: admin_app/src/app/dashboard/components/TabPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import Tab from "@mui/material/Tab";
+import Box from "@mui/material/Box";
+
+interface TabPanelProps {
+  tabValue: number;
+  handleChange: (event: React.SyntheticEvent, newValue: number) => void;
+}
+
+const TabPanel: React.FC<TabPanelProps> = ({ tabValue, handleChange }) => {
+  const tabLabels = ["Last 24 hours", "Last week", "Last month", "Last year"];
```
Comment (lickem22) [inline admin_app/src/app/dashboard/components/TabPanel.tsx:12]: Shouldn't this be in `index.ts rather than here

File: admin_app/src/app/playground/layout.tsx [hunk 5]
```diff
@@ -9,8 +10,14 @@ export default function RootLayout({
 }>) {
   return (
     <ProtectedComponent>
-      <NavBar />
-      {children}
+      <Grid container spacing={3}>
```
Comment (lickem22) [inline admin_app/src/app/playground/layout.tsx:13]: Why this change?

File: admin_app/src/components/NavBar.tsx (hunk 2 above)
Comment (lickem22) [reply admin_app/src/components/NavBar.tsx:31]: Oh okay thanks, answered my question on `urgency-rules/layout.tsx`

**Review by lickem22 — APPROVED** [review]
Thanks @sidravi1, looks promising. No major comments so I approve!

File: admin_app/src/app/dashboard/components/TabPanel.tsx (hunk 4 above)
Comment (sidravi1) [reply admin_app/src/app/dashboard/components/TabPanel.tsx:12]: I'll end up moving this when I build the page. Thanks!

File: admin_app/src/app/dashboard/components/Overview.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        <Box
+          bgcolor="white"
+          sx={{
+            flexGrow: 0,
+            borderRadius: 1,
+            width: 350,
+            height: 400,
+          }}
+        >
+          Heatmap
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/app/dashboard/components/Overview.tsx:76]: Sounds good. Will fix in the next PR.
